api/app/signals/apps/questionnaires/services/questionnaires.py

```diff
@@ -112,6 +112,14 @@ def create_answer(answer_payload, question, questionnaire, session=None):
 
         return answer
 
+    @staticmethod
+    def freeze_session(session):
+        session.frozen = True
+        session.save()
+
+        QuestionnairesService.handle_frozen_session(session)
```

Comment:
What happens to the Session if this functionality raises an exception? For example in the ReactionRequestService.handle_frozen_session_REACTION_REQUEST a SessionNotFrozen or a WrongFlow exception can be raised. Do we keep the Session frozen and/or raise an exception?